### PR TITLE
Cargo command editor

### PR DESCRIPTION
**Pull request by matklad**
Add UI component for adding Cargo commands with completion. Currently it is used only in **Run Cargo Command** action, but the plan is to plug it into run configurations as well.

https://zippy.gfycat.com/FixedGiantAiredaleterrier.webm

r? @vlad20012

### CODE REVIEW

File: src/main/kotlin/org/rust/cargo/util/CargoCommandCompletionProvider.kt [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+
+    override fun acceptChar(c: Char): CharFilter.Result? =
+        if (c == '-') CharFilter.Result.ADD_TO_PREFIX else null
+
+    override fun addCompletionVariants(text: String, offset: Int, prefix: String, result: CompletionResultSet) {
+        val (ctx, _) = splitContextPrefix(text)
+        result.addAllElements(complete(ctx))
+    }
+
+    // public for testing
```
Comment (mkaput) [inline src/main/kotlin/org/rust/cargo/util/CargoCommandCompletionProvider.kt]: https://upsource.jetbrains.com/idea-ce/file/idea-ce-200f59d7cf3b0f66feb3a4abebbb90864dc5edc7/platform/annotations/common/src/org/jetbrains/annotations/TestOnly.java

Comment (matklad) [inline src/main/kotlin/org/rust/cargo/util/CargoCommandCompletionProvider.kt]: Hm, this is called not only in tests, but in implementation as well. So what we really want here is https://google.github.io/guava/releases/19.0/api/docs/com/google/common/annotations/VisibleForTesting.html, but Guava annotations are not generally used in IntelliJ.

Comment (vlad20012) [inline src/main/kotlin/org/rust/cargo/util/CargoCommandCompletionProvider.kt]: You can make it private and add separate TestOnly accessors

Comment (mkaput) [inline src/main/kotlin/org/rust/cargo/util/CargoCommandCompletionProvider.kt]: > Hm, this is called not only in tests, but in the implementation as well. So what we really want here is https://google.github.io/guava/releases/19.0/api/docs/com/google/common/annotations/VisibleForTesting.html, but Guava annotations are not generally used in IntelliJ.

Didn't read carefully what `@TestOnly` means, heh

> You can make it private and add separate TestOnly accessors

Nah, too much hassle 😉

Comment (matklad) [inline src/main/kotlin/org/rust/cargo/util/CargoCommandCompletionProvider.kt]: Hm, generally this sounds like an excellent approach, but I am afraid I am too lazy in the end :)

**Comment by matklad** [comment]
bors r=vlad20012

**Pull request merged**